lib/plugins/aws/package/compile/events/apiGateway/lib/deployment.js

```diff
@@ -14,6 +14,7 @@ module.exports = {
         Properties: {
           RestApiId: this.provider.getApiGatewayRestApiId(),
           StageName: this.provider.getStage(),
+          Description: this.provider.getApiGatewayDescription(),
```

Comment:
Speaking up because I thing this feature will actually fix https://github.com/serverless/serverless/issues/3495#issuecomment-438251252 is implemented correctly. 

The cloud formation resource for this is not a string but a structured type 
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-deployment.html

It's the structured type that enables turning on active tracing for the gateway https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-deployment-stagedescription.html

Reply:
I take that back. That's StageDescription vs Description. You may get more bang for your buck if you add them both here. I'd but happy to add a follow pull that just mirrors this with StageDescription if that's too much scope.

Reply:
Yes, I can add `StageDescription` - it's not a problem.
I think would be better if I just extend this PR where `StageDescription` also will be supported.
What do you say about that ?

Reply:
I'd prefer if it only sets the description if it's provided instead of setting it to an empty string if it's not

Reply:
@dschep I've done